### PR TITLE
Login auth

### PR DESCRIPTION
**Pull request by kart-c**
### Login and Signup Authorization

Changes made in this PR
- Implement Login Auth
- Implement Signup Auth
- User is directed to a new user page if already logged in

> Note: 
> - Inline style in user page is temporary. This will be changed when a user page feature is created. 
> - All alerts are intentional and will be changed later on.

### CODE REVIEW

**Review by VishalPatil18 — APPROVED** [review]

File: src/Pages/Signup/Signup.jsx [hunk 1]
```diff
@@ -1,43 +1,127 @@
...
+						authDispatch({
+							type: 'SIGNUP',
+							payload: { token: response.data.encodedToken, user: response.data.createdUser },
+						});
+						localStorage.setItem('token', response.data.encodedToken);
+						localStorage.setItem('user', JSON.stringify(response.data.createdUser));
+						navigate('/');
+						alert('Signed up');
+					} else {
+						console.error('ERROR: ', response);
```
Comment (RahullRawat) [inline src/Pages/Signup/Signup.jsx:42]: You can remove this console.log

File: src/Pages/Signup/Signup.jsx [hunk 2]
```diff
@@ -1,43 +1,127 @@
...
+					}
+				} catch (error) {
+					console.error(error);
```
Comment (RahullRawat) [inline src/Pages/Signup/Signup.jsx:45]: This

File: src/Pages/Login/Login.jsx [hunk 3]
```diff
@@ -1,36 +1,116 @@
...
+				localStorage.setItem('user', JSON.stringify(response.data.foundUser));
+				navigate('/');
+				alert('Logged in');
+			} else {
+				console.error('ERROR: ', response);
+				alert('ERROR');
+			}
+		} catch (error) {
+			alert('ERROR');
+			console.error(error);
```
Comment (RahullRawat) [inline src/Pages/Login/Login.jsx:56]: This

File: src/Pages/Login/Login.jsx [hunk 4]
```diff
@@ -1,36 +1,116 @@
...
+				authDispatch({
+					type: 'LOGIN',
+					payload: { token: response.data.encodedToken, user: response.data.foundUser },
+				});
+				localStorage.setItem('token', response.data.encodedToken);
+				localStorage.setItem('user', JSON.stringify(response.data.foundUser));
+				navigate('/');
+				alert('Logged in');
+			} else {
+				console.error('ERROR: ', response);
```
Comment (RahullRawat) [inline src/Pages/Login/Login.jsx:51]: This

**Review by RahullRawat — APPROVED** [review]
Very well written code, easy to understand. Great Work @kart-c

**Review by mohammed-sas — APPROVED** [review]
Well structured code.
well written auth-context and the associated reducer
persisting user info despite page refresh is good

File: src/Pages/Signup/Signup.jsx (hunk 1 above)
Comment (kart-c) [reply src/Pages/Signup/Signup.jsx:42]: These are intentional console.logs for showing error, if there is any, to the reviewer. The one's mentioned below are also intentional.

**Pull request merged**